pkg/object/mutation/types.go

```diff
@@ -124,18 +124,32 @@ func (r ResourceReference) ObjMetadata() object.ObjMetadata {
 	}
 }
 
-// Unstructured returns the name, namespace, group, version, and kind of the
+// ToUnstructured returns the name, namespace, group, version, and kind of the
 // ResourceReference, wrapped in a new Unstructured object.
 // This is useful for performing operations with
 // sigs.k8s.io/controller-runtime/pkg/client's unstructured Client.
-func (r ResourceReference) Unstructured() *unstructured.Unstructured {
+func (r ResourceReference) ToUnstructured() *unstructured.Unstructured {
 	obj := &unstructured.Unstructured{}
 	obj.SetName(r.Name)
 	obj.SetNamespace(r.Namespace)
 	obj.SetGroupVersionKind(r.GroupVersionKind())
 	return obj
 }
 
+// ToUnstructured returns the name, namespace, group, and kind of the
+// ResourceReference, wrapped in a new ObjMetadata object.
+func (r ResourceReference) ToObjMetadata() object.ObjMetadata {
+	gk := r.GroupVersionKind().GroupKind()
+	return object.ObjMetadata{
+		Namespace: r.Namespace,
+		Name:      r.Name,
+		GroupKind: schema.GroupKind{
+			Group: gk.Group,
+			Kind:  gk.Kind,
+		},
+	}
```

Comment:
nit: here and in a few other places:
```suggestion
	return object.ObjMetadata{
		Namespace: r.Namespace,
		Name:      r.Name,
		GroupKind: r.GroupVersionKind().GroupKind(),
	}
```

Reply:
I actually explicitly created a new GroupKind in order to avoid pointing to one in the object being copied from (and sharing modifications). But now that I think about it, since it's not a pointer, it's probably doing a copy-by-value on the whole GroupKind struct. So the condensed version is probably safe.

Reply:
fixed.